pkg/web_app/lib/src/screenshot_carousel.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:html';
+
+void setupScreenshotCarousel() {
+  _setEventForScreenshot();
+}
+
+void _setEventForScreenshot() {
+  final carousel = document.getElementById('-screenshot-carousel');
+  if (carousel != null) {
+    final thumbnails = document.querySelectorAll('.screenshot-thumbnail');
+    final imageContainer = document.getElementById('-image-container')!;
+    final prev = document.getElementById('-carousel-prev')!;
+    final next = document.getElementById('-carousel-next')!;
+    ImageElement? imageElement =
+        document.getElementById('-carousel-image') as ImageElement?;
+    if (imageElement == null) {
+      imageElement = ImageElement();
+      imageElement.id = '-carousel-image';
+      imageContainer.children.add(imageElement);
+      imageElement.className = 'carousel-image';
+    }
+
+    List<String> images = [];
+
+    void hideElement(Element element) {
+      element.style.display = 'none';
+    }
+
+    void showElement(Element element) {
+      element.style.display = 'flex';
+    }
+
+    void showImage(int index, UIEvent event) {
+      event.stopPropagation();
+      imageElement!.src = images[index];
+      if (index == images.length - 1) {
+        hideElement(next);
+      }
+      if (index == 0) {
+        hideElement(prev);
+      }
+      if (index > 0) {
+        showElement(prev);
+      }
+      if (index < images.length - 1) {
+        showElement(next);
+      }
+    }
+
+    int screenshotIndex = 0;
+    for (final thumbnail in thumbnails) {
+      thumbnail.onClick.listen((event) {
+        showElement(carousel);
+        images = thumbnail.attributes['data']!.split(',');
+        showImage(screenshotIndex, event);
+      });
+    }
+
+    prev.onClick.listen((event) {
+      showImage(--screenshotIndex, event);
+    });
+
+    next.onClick.listen((event) {
+      showImage(++screenshotIndex, event);
+    });
+
+    imageElement.onClick.listen((event) {
+      event.stopPropagation();
+    });
+
+    void closeCarousel() {
+      hideElement(carousel);
+      hideElement(next);
+      hideElement(prev);
+      screenshotIndex = 0;
+    }
+
+    carousel.onClick.listen((event) {
+      event.stopPropagation;
+      closeCarousel();
+    });
+
+    document.onKeyDown.listen((event) {
+      if (carousel.style.display != 'none') {
+        event.stopPropagation();
+        if (event.key == 'Escape') {
+          closeCarousel();
+        }
+        if (event.key == 'ArrowLeft') {
+          if (screenshotIndex > 0) {
+            showImage(--screenshotIndex, event);
+          }
+        }
+        if (event.key == 'ArrowRight') {
+          if (screenshotIndex < images.length - 1) {
+            showImage(++screenshotIndex, event);
+          }
+        }
```

Comment:
Only handle event when taking action.
```suggestion
        if (event.key == 'Escape') {
          closeCarousel();
          event.stopPropagation();
        }
        if (event.key == 'ArrowLeft') {
          if (screenshotIndex > 0) {
            showImage(--screenshotIndex, event);
            event.stopPropagation();
          }
        }
        if (event.key == 'ArrowRight') {
          if (screenshotIndex < images.length - 1) {
            showImage(++screenshotIndex, event);
            event.stopPropagation();
          }
        }
```

Reply:
`showImage` already handles the event. Adding it to `closeCarousel` for consistency.